pkg/plugin/sdk/v1alpha2/plugin/grpc.go

```diff
@@ -74,6 +75,46 @@ func (m *GRPCServer) PostBuildHook(
 		m.Impl.PostBuildHook(req.IsInteractiveMode, prompter)
 }
 
+// CustomCommands translates the gRPC call to the Plugin CustomCommands
+// implementation. It returns a list of commands that the plugin implements.
+func (m *GRPCServer) CustomCommands(
+	ctx context.Context,
+	req *proto.CustomCommandsReq,
+) (*proto.CustomCommandsRes, error) {
+	customCommands, err := m.Impl.CustomCommands()
+
+	saveRunFunctions(customCommands)
+
+	pbCommands := make([]*proto.Command, 0)
+	for _, command := range customCommands {
+		pbCommand := &proto.Command{
```

Comment:
The issue I have with this is that when a plugin wants to return a `Command` they will be forced to add a `Command: &proto.Command{}` to their `Command` since you cannot currently set embedded struct values directly. Would look like this: 
```
&plugin.Command{
		Command: &proto.Command{
			Use:       "foo",
			ShortDesc: "bar",
			LongDesc:  "bar long",
		},
		Run: func(ctx context.Context, args []string) error {
			return nil
		},
	}
```

I much prefer how it currently looks:
```
&plugin.Command{
		Use:       "foo",
		ShortDesc: "bar",
		LongDesc:  "bar long",
		Run: func(ctx context.Context, args []string) error {
			return nil
		},
	}
```

Reply:
I feel like the second option creates a much better user experience.

Reply:
```go
func NewCommand(
    use string,
    shortDesc string,
    longDesc string,
    run func(ctx context.Context, args []string) error,
) *plugin.Command {
    return &plugin.Command{
        Command: &proto.Command{
            Use: use,
            ShortDesc: shortDesc,
            LongDesc: longDesc,
        },
        Run: run,
    }
}
```

Reply:
Added to the interface file so that its easy to use